pkg/operator/ceph/csi/secrets.go

```diff
@@ -183,26 +228,28 @@ func CreateCSISecrets(context *clusterd.Context, clusterInfo *client.ClusterInfo
 	}
 	k := keyring.GetSecretStore(context, clusterInfo, clusterInfo.OwnerInfo)
 
+	csiCephXConfig := clusterSpec.Security.CephX.CSI
+
 	// Create CSI RBD Provisioner Ceph key
-	csiRBDProvisionerSecretKey, err := createCSIKeyringRBDProvisioner(k)
+	csiRBDProvisionerSecretKey, _, err := createCSIKeyring(context, clusterInfo, k, csiCephXConfig, csiKeyringRBDProvisionerUsername, cephCSIKeyringRBDProvisionerCaps())
```

Comment:
Let's not ignore the key count. Instead, record all of them for the various 4 keys, and at the end, pick the `max()` of all of them to use for processing. At the end, if the max count is greater than the user's desired max count, we can return an error to re-reconcile and get to the user's desired state in the spec config.